packages/blade/src/components/Tooltip/_decisions/decisions.md

```diff
@@ -0,0 +1,198 @@
+
+```js
+// non-interactive element as trigger
+<Tooltip content="Amount reversed to customer bank account" shouldWrapNonInteractiveElement>
+  <Counter value={100} />
+</Tooltip>
+```
+
+> Note: If users don't wrap the non-interactive elements the tooltip won't apear
+
+### Custom Triggers
+
+Users will be able to use their own custom components as triggers:
+
+To make custom elements work the components needs to expose:
+
+- ref
+- Web:
+  - onBlur
+  - onFocus
+  - onMouseLeave
+  - onMouseMove
+  - onPointerDown
+  - onPointerEnter
+- Native:
+  - onTouchStart
+  - onTouchEnd
+
+```jsx
+import type { TooltipTriggerProps } from "@razorpay/blade/components";
+
+type MyCustomButtonProps = {} & TooltipTriggerProps
+const MyCustomButton = React.forwardRef(...);
+
+// custom trigger
+<Tooltip content="Amount reversed to customer bank account" shouldWrapNonInteractiveElement>
+  <MyCustomButton>Click me<MyCustomButton>
+</Tooltip>
+```
+
+## Library
+
+We will be using [FloatingUI](https://floating-ui.com/) to position the tooltip & handle the basic tooltip logic. Floating UI supports both Web & ReactNative.
+
+### Bundle Size
+
+With all the neccessary middlewares and features included for our usecase FloatingUI will be around **10kb** gzip.
+
+Check the [BundleJS Analysis](https://bundlejs.com/?q=%40floating-ui%2Freact&treeshake=%5B%7B%0A++arrow%2Cflip%2CFloatingArrow%2Coffset%2CuseFloating%2CuseFocus%2CuseHover%2CuseInteractions%2CuseTransitionStyles%2C%7D%5D&config=%7B%22esbuild%22%3A%7B%22external%22%3A%5B%22react%22%2C%22react-dom%22%5D%7D%7D)
+
+This is worth the size becaue:
+
+- FloatingUI supports both react / react-native
+- It handles
+  - Animations (mount/unmount)
+  - Automatic collision aware placements
+  - Interactions (hover, focus)
+  - Tooltip Arrow
+  - Middlewares
+
+<details>
+  <summary>Implementation detail nuances:</summary>
+
+To make FloatingUI work with Blade components seamlessly there are few things we need to modify:
+
+1. Expose interaction props in all the trigger components
+
+- Web:
+  - onBlur
+  - onFocus
+  - onMouseLeave
+  - onMouseMove
+  - onPointerDown
+  - onPointerEnter
+- Native:
+  - onTouchStart
+  - onTouchEnd
+
+2. Expose the actual DOM node from ref instead of only exposing [certain methods via our useBladeInnerRef](https://github.com/razorpay/blade/blob/69a1bcef2f09ceaf6f910eaaca3076055fb059a2/packages/blade/src/hooks/useBladeInnerRef.web.ts#L26-L27) hook which we used to prevent component styling misuses. We need to expose the actual DOM node because FloatingUI internally does this [isElement() check](https://github.com/floating-ui/floating-ui/blob/b8990250568043b876e1c8fe42358fe337847ede/packages/react/src/hooks/useFloating.ts#L59) on the tooltip trigger element.
+
+</details>
+
+## Motion
+
+TBD
+
+## Accessibility
+
+- Tooltip will have [`role=tooltip`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role)
```

Comment:
we don't need `aria-labelledby` and `aria-describedby` attrs? because on icons you might need those things for screen readers?

Reply:
Yes added.

Reply:
can't find it? 🤔

Reply:
pushed check again? 
<img width="751" alt="image" src="https://github.com/razorpay/blade/assets/35374649/0047057b-a5ba-43ef-8f95-4f3453db9670">